### PR TITLE
fix: require no partial vault deposits

### PR DESCRIPTION
**Pull request by dudesahn**
Vault code that will prevent partial deposits: https://github.com/yearn/yearn-vaults/blob/97ca1b2e4fcf20f4be0ff456dabd020bfeb6697b/contracts/Vault.vy#L923

### CODE REVIEW

**Comment by storming0x** [comment]
weird CI test is failing, can you paste the screen shot?

File: tests/test_zap_fix.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    # oChad lowers deposit yvdai, fucking rude!
+    yvdai.setDepositLimit(yvdai.totalAssets() + 100e18, {"from": gov})
+
+    # update our zap contract
+    yvdai_pool_live.setZapContract(new_zap, {"from": gov})
+
+    # zap in, but it should fail
+    with brownie.reverts():
+        new_zap.zapIn(yvdai, dai_amount, {"from": dai_whale})
+    assert dai.balanceOf(dai_whale) == dai_starting
```
Comment (storming0x) [inline tests/test_zap_fix.py:79]: good to add zap doesnt have any balance at the end assert

**Review by storming0x — APPROVED** [review]
changes LGTM

**Review by rareweasel — APPROVED** [review]
LGTM

**Comment by rareweasel** [comment]
Changes LGTM.

It looks like some configuration is not working in the ci. Please, if you can, share some screenshot with the tests passing.

**Comment by dudesahn** [comment]
hey guys, coming back to this—CI is failing (I believe) because of optimism not being naturally built in to brownie, so it won't recognize the network for it. here's a screenshot of the tests passing after I added @storming0x additional asserts. will add deployed address in a bit
Screen Shot 2023-05-30 at 8 19 13 PM

**Comment by dudesahn** [comment]
### Deployed Contract
- [StakingRewardsZap](https://optimistic.etherscan.io/address/0x498d9dcbb1708e135bdc76ef007f08cba4477be2)

**Pull request merged**